### ISSUE TITLE
[Serve] Ray Serve leaking replicas in rolling update

### ISSUE THREAD

**Issue by jiaodong** [ray-project/ray#24260]
### What happened + What you expected to happen

From user reported cc: @frosk1 

This script simply leads to a memory overflow:

```
@serve.deployment(route_prefix="/test", name="testdeployment")
class TesterDepl:
  def __init__(self):
      self.count = 0
  def __call__(self):
      self.count += 1
      return self.count

ray.init(address="auto")
serve.start(detached=True)
while True:
    TesterDepl.deploy()
```

When calling deploy() again, shouldn't this delete all further resources (serve replicas) and instantiate a new one?
Instead it just creates a new replica of the same deployment again and again  filling up memory till OOM error .

Calling 

```
ray.init(address="auto")
ray.serve.serve.start()
ray.serve.serve.shutdown()
```

this kills the SERVE_CONTROLLER_ACTOR and the SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-node
but the SERVE_REPLICAs  are still alive afterwards.

### Versions / Dependencies

Ray 1.12

### Reproduction script

.

### Issue Severity

High: It blocks me from completing my task.

**Comment by frosk1** [comment]
After testing and input from Shreyas Krishnaswamy  *sry couldnt find the github handle from him :)

it seems to be a namespace issue. This here resolves the problem:

`ray.init(address="auto", namespace="serve")`

**Comment by jiaodong** [comment]
Thanks for the confirmation ! Looks like we have this issue on Ray 1.12 and need a bit more investigation and decide if we need to make a cherry-pick release. @shrekris-anyscale i've assigned this to you for now to see if the fix is sufficient or if we need any other next steps

**Comment by jiaodong** [comment]
- We confirmed it's reproducible on 1.12, not on 1.11 or master.
- Shreyas is checking into which PR might have it fixed and verify locally against 1.12.1 branch
- We suggest finding the fix and cherry-pick on 1.12 asap and started a thread for it.
